sdks/java/extensions/sorter/README.md

```diff
@@ -0,0 +1,44 @@
+<!--
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+-->
+
+Sorter
+============
+
+The provided Sorter can be used to sort large (more than can fit to memory)
```

Comment:
Make the focus of the documentation `SortValues` rather than `Sorter` -- `SortValues` is the primary user-facing object here. E.g:

This module provides the `SortValues` transform, which takes a `PCollection<KV<K,Iterable<V>>>` and produces a `PCollection<KV<K,Iterable<V>>>` where the `Iterable<V>` has been sorted for each key. It uses various `Sorter` implementations to efficiently and scalably sort the iterables, even if they are large (do not fit in memory).

## Caveats

* This transform performs *value-only sorting*; the iterable accompanying each key is sorted, but there is *no relationship between different keys*, as Beam does not support any defined relationship between different elements in a `PCollection`.
* For current `Sorter` implementations, each `Iterable<V>` is sorted on a single worker using local memory and disk. This means that `SortValues` may be a performance and/or scalability bottleneck when used in different pipelines. For example, users are discouraged from using `SortValues` on a `PCollection` of a single element to globally sort a large `PCollection`.

## `Sorter` implementations

* `InMemorySorter` implements sorting in memory and will (do what, fail?) if the input `Iterable<V>` is too large.
* `ExternalSorter` implements sorting using a local disk (in `/tmp` by default`).
* `BufferedExternalSorter` is the recommended `Sorter` to be used with `SortValues`. It uses `InMemorySorter` for a small iterable, and falls back to `ExternalSorter` when the iterable to be sorted exceeds a user-set threshold.

## Using `SortValues`

the code you have now.